Disable encoding of &, <, > characters in RSF item

chr1sdi opened the pull request:
https://trello.com/c/Xhq7HJsn/617-serializer-producing-non-canonical-json-containing-escaped-unicode-characters

Those characters were encoded by default (HTML encoding) as some browsers
may have problems with them (explanation from GO lang). We had to switch
that off as RSF generated like that was rejected as it's not canonical
version.

Additionally new implementation of encoder seem to leave some whitespaces
and after looking at stream_test.go where encoders with HTML encoding switched
off are tested we implemented a similar solution which is to trim whitespaces.
This is really terrible on many levels and has ruined our afternoon.

chr1sdi committed:
Disable encoding of &, <, > characters in RSF item

Those characters were encoded by default (HTML encoding) as some browsers
may have problems with them (explanation from GO lang). We had to switch
that off as RSF generated like that was rejected as it's not canonical
version.

Additionally new implementation of encoder seem to leave some whitespaces
and after looking at stream_test.go where encoders with HTML encoding switched
off are tested we implemented a similar solution which is to trim whitespaces.
This is really terrible on many levels and has ruined our afternoon.

sophiebrown01 commented on serializer.go:
```diff
@@ -67,7 +67,7 @@ func getKey(fieldNames []string, fieldValues []string, registerName string) (str
 
 func processEmptyRootHash() {
-	fmt.Println("assert-root-hash\te3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
+	fmt.Println("assert-root-hash\tsha256:e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855")
```
This (the hashing algorithm) was missing before, so we added it.

sophiebrown01 commented on utils.go:
```diff
@@ -49,8 +49,12 @@ func toJsonArrayOfNum(s string) string {
-	return string(data), err
+	var buf bytes.Buffer
+	enc := json.NewEncoder(&buf)
+	enc.SetEscapeHTML(false)
```
This is the part that actually disables encoding `&, <, >`.

johnollier commented on utils.go:
```diff
@@ -49,8 +49,12 @@ func toJsonArrayOfNum(s string) string {
+	err := enc.Encode(r)
+
+	return strings.TrimSpace(buf.String()), err
 }
```
According to the docs here https://golang.org/pkg/encoding/json/#Encoder.Encode you get a bonus newline character. So if we are concerned that trimming white space might change the actual content, maybe we could use this:

strings.TrimSuffix(buf.String(),"\n")

The tests still pass.

chr1sdi committed:
Explicitly trim the new line suffix from encoded JSON

As per documentation after encoding the new line is added to
the string, we have to trim it as we don't need that happening.

chr1sdi committed:
Ignore IDE files in gitignore
